### PR TITLE
Tons of fixes for almost each part of the code

### PR DESCRIPTION
**Pull request by zpelechova**

### CODE REVIEW

File: INPUT_SCHEMA.json [hunk 1]
```diff
@@ -2,14 +2,29 @@
     "title": "Autotrader Scraper Input Schema",
     "type": "object",
     "schemaVersion": 1,
+    "description": "(Optional if 'predefined filters' are enabled) Search is based on URL query string. You can insert multiple URLs. Filters must be in the query string, each URL query string can contain any allowed filter. Check Autotrader advanced search and/or actor documentation for more details.",
     "properties": {
+        "startUrls": {
+            "title": "Start Urls",
+            "type": "array",
+            "editor": "requestListSources",
+            "description": "(Optional if 'predefined filters' are enabled) List of search URLs to start the scraping. It must be a search url including querystring for filters. Check documentation for details and examples",
```
Comment (pocesar) [inline INPUT_SCHEMA.json:11]: make sure to test the INPUT_SCHEMA using `apify vis` command, before merging

File: INPUT_SCHEMA.json [hunk 2]
```diff
@@ -2,14 +2,29 @@
...
         "enablePredefinedFilters": {
             "title": "Enable this section",
             "type": "boolean",
             "description": "(Required if 'startUrls' is not provided) Enable predefined filter section",
-            "sectionCaption": "Search with predefined filters (Single search)",
-            "sectionDescription": "Set filters for search",
-            "default": true
+            "sectionCaption": "Filter cars here (instead of URL)",
+            "sectionDescription": "Set filters for search. Be aware that you can either use these or URL, not both.",
+            "default": false
```
Comment (pocesar) [inline INPUT_SCHEMA.json:27]: is this a breaking change?

File: package.json [hunk 3]
```diff
@@ -3,9 +3,9 @@
 	"version": "0.0.1",
```
Comment (pocesar) [inline package.json]: add the `devDependencies` here, so you can lint and fix errors before committing 
```suggestion
	"version": "0.0.1",
	"devDependencies": {
            "@apify/eslint-config": "^0.2.2",
            "eslint": "^8.7.0"
         },
```

the source files can be fixed using this script command:

```jsonc
    "scripts": {
        "start": "node src/main.js",
        "lint": "./node_modules/.bin/eslint ./src --ext .js,.jsx", 
        "lint:fix": "./node_modules/.bin/eslint ./src --ext .js,.jsx --fix",
        "test": "echo \"Error: oops, the actor has no tests yet, sad!\" && exit 1"
    },
```
run using `npm run lint:fix`

File: src/utils.js [hunk 4]
```diff
@@ -1,174 +1,221 @@
...
+    userResult = await evaledFunc($)
+  } catch (err) {
+    log.error(
+      `extendOutputFunction crashed! Pushing default output. Please fix your function if you want to update the output. Error: ${err}`
+    )
+  }
+
+  if (!isObject(userResult)) {
+    log.exception(new Error('extendOutputFunction must return an object!'))
+    process.exit(1)
```
Comment (pocesar) [inline src/utils.js]: don't use `process.exit(1)` as it won't give enough time for the scraper to save stuff to the KV or RequestQueue

File: src/utils.js [hunk 5]
```diff
@@ -1,174 +1,221 @@
...
-    if (itemCount >= maxItems) {
-        log.info('Actor reached the max items limit. Crawler is going to halt...');
-        log.info('Crawler Finished.');
-        process.exit();
-    }
+function maxItemsCheck (maxItems, itemCount) {
+  if (itemCount >= maxItems) {
+    log.info('Actor reached the max items limit. Crawler is going to halt...')
+    log.info('Crawler Finished.')
+    process.exit()
```
Comment (pocesar) [inline src/utils.js]: prefer using `crawler.autoscaledPool.abort()`

**Review by pocesar — CHANGES_REQUESTED** [review]
thanks @zpelechova! some preliminary comments. 
when editing code, try not to change the formatting, it can be hard to keep track of changes. whitespace (new lines, indentation), `;`s and moving code up and down without changing it. 
make sure to always use `eslint` (and having the vscode extension installed) to spot some issues while coding. after that you can use the `npm run lint` command and it will show you the lines that are "out of style". some of those can be fixed with `npm run lint:fix`

File: INPUT_SCHEMA.json (hunk 1 above)
Comment (zpelechova) [reply INPUT_SCHEMA.json:11]: Input schema is valid (response).

File: INPUT_SCHEMA.json (hunk 2 above)
Comment (zpelechova) [reply INPUT_SCHEMA.json:27]: Well, the actors hasnt worked at all for some time, so I dont think we have to worry about that. I changed it bcs of the logic of the input schema, would do the same if I only changed that...

File: src/utils.js (hunk 4 above)
Comment (zpelechova) [reply src/utils.js]: What else to use? Althought I dont think I added this there, I must have just changed the eslint on it.

Comment (zpelechova) [reply src/utils.js]: Oh I see, the think down!

Comment (zpelechova) [reply src/utils.js]: Right? crawler.autoscaledPool.abort() ?

File: package.json (hunk 3 above)
Comment (zpelechova) [reply package.json]: Cannot make it work: 
image

**Comment by pocesar** [comment]
@zpelechova added the lint and fixed the style issues

File: src/main.js [hunk 6]
```diff
@@ -82,69 +81,117 @@ Apify.main(async () => {
...
+                    $('span:contains(Results)')
+                        .text()
+                        .replace(/\D/g, ''),
+                ),
+                );
+
+                // split urls by price filter
+                if (totalResults > 1000) {
+                    log.info(
+                        'Too many results for one scrape, splitting the search into two.',
```
Comment (metalwarrior665) [inline src/main.js:133]: Logs should always include info about the current URL and you could even add how you are gonna split it.

File: src/main.js [hunk 7]
```diff
@@ -82,69 +81,117 @@ Apify.main(async () => {
...
+                console.log(searchUrl);
+
                 const info = await Apify.utils.enqueueLinks({
                     page,
                     requestQueue,
                     selector: 'div.display-flex.justify-content-between > a',
-                    pseudoUrls: ['https://www.autotrader.com/cars-for-sale/vehicledetails.xhtml?listingId=[.*]'],
+                    pseudoUrls: [
+                        'https://www.autotrader.com/cars-for-sale/vehicledetails.xhtml?listingId=[.*]',
+                    ],
```
Comment (metalwarrior665) [inline src/main.js:164]: No sure what formatting you use but the previous version is better I think

**Review by metalwarrior665 — COMMENTED** [review]
Just small comments

**Comment by metalwarrior665** [comment]
@zpelechova @pocesar We can merge?

**Review by pocesar — APPROVED** [review]

**Pull request merged**